rest-service/manager_rest/resource_manager.py

```diff
@@ -1057,6 +1061,76 @@ def add_node_instance_from_dict(self, instance_dict):
         # adding it to the response.
         instance_dict['resource_availability'] = resource_availability
 
+    def list_agents(self, deployment_id=None, node_ids=None,
+                    node_instance_ids=None, install_method=None):
+        filters = {}
+        if deployment_id is not None:
+            filters['deployment_id'] = deployment_id
+        if node_ids is not None:
+            filters['id'] = node_ids
+        nodes = {n._storage_id: n
+                 for n in self.sm.list(models.Node, filters=filters,
+                                       get_all_results=True)
+                 if self._is_agent_node(n)}
+
+        if not nodes:
+            pagination = {'total': 0, 'size': 0, 'offset': 0}
+            return ListResult([], metadata={'pagination': pagination})
+
+        instance_filters = {'_node_fk': list(nodes)}
+        if node_instance_ids:
+            instance_filters['id'] = node_instance_ids
+        node_instances = self.sm.list(models.NodeInstance,
+                                      filters=instance_filters,
+                                      get_all_results=True)
```

Comment:
What will this do in the case of there being more than max_results node instances?